src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/MergedCalController.java

```diff
@@ -18,8 +26,31 @@
   @Autowired
   MergedCalendarRepository mergedCalendarRepository;
 
+  @Autowired
+  UserModelRepository userModelRepository;
+
+  @Autowired
+  AuthModelRepository authModelRepository;
+
   @PostMapping(value = "/calendar")
-  public ResponseEntity postMergedCal(@RequestBody MergedCalendar mergedCalendar) {
+  public ResponseEntity postMergedCal(@RequestHeader("X-Client-Token") String clientToken, @RequestBody MergedCalendarFromAndroid mergedCalendarFromAndroid) {
+    MergedCalendar mergedCalendar = new MergedCalendar();
+    //mergedCalendar.setUser(userModelRepository.findByClientToken(clientToken));
+    mergedCalendar.setOutputCalendarId(mergedCalendarFromAndroid.getOutputCalendarId());
+    String[] inputStrings = mergedCalendarFromAndroid.getInputCalendarIds();
+
+    List<CalendarId> calendarIds = new ArrayList<>();
+    for (int i = 0; i < inputStrings.length; i++) {
+      CalendarId calendarId = new CalendarId();
+      calendarId.setId(inputStrings[i]);
+      calendarId.setAuthModel(null);
+      calendarId.setMergedCalendar(null);
+      calendarIds.add(calendarId);
+    }
+    mergedCalendar.setCalendarId(calendarIds);
+    mergedCalendar.setUserName(userModelRepository.findByClientToken(clientToken).getUserEmail());
+    mergedCalendar.setOutputAccount(userModelRepository.findByClientToken(clientToken).getUserEmail());
+   //mergedCalendar.setUser(null);
     mergedCalendarRepository.save(mergedCalendar);
     return new ResponseEntity(HttpStatus.OK);
```

Comment:
I hope you will have time to refactor this code before the sponsor demo.  Most of the things above should got to a service class to service methods, and no outcommented code please!